cli/listener/src/main.rs

```diff
@@ -138,24 +138,51 @@ pub fn get_apy_for_period(
             slot: row.get(2)?,
             epoch: row.get(3)?,
             pool: row.get(4)?,
-            token_a: row.get(5)?,
-            token_b: row.get(6)?,
+            price_lamports_numerator: row.get(5)?,
+            price_lamports_denominator: row.get(6)?,
         })
     };
 
-    // Get first logged minimal logged data based on timestamp that is greater than `from_time`.
-    let mut stmt_first = conn.prepare(
-        "SELECT *, MIN(timestamp) FROM exchange_rate WHERE pool = :pool AND timestamp > :t",
-    )?;
-    let mut row_iter = stmt_first.query_map([opts.pool.clone(), from_time.to_string()], row_map)?;
-    let first = row_iter.next();
+    let (first, last) = {
+        // Get first logged minimal logged data based on timestamp that is greater than `from_time`.
+        let stmt_first = &mut tx.prepare(
+            "WITH prices_epoch AS (
+                SELECT *
+                FROM exchange_rate
+                WHERE epoch = (SELECT MIN(epoch) from exchange_rate where pool = :pool AND timestamp > :t)
+              )
+              SELECT
+                *
+              FROM
+                prices_epoch
+              ORDER BY
+                timestamp ASC
```

Comment:
Small note: we might add a `LIMIT 1` as well. It wouldn’t affect the outcome as we only run `.next()` once anyway, but I can imagine it might affect the query plan. For now I wouldn’t bother changing it and if we ever find that our queries are too slow we can investigate.

Reply:
Yeah... Also we might need to change it to be the first entries of epoch + 200. I'll put a TODO

Reply:
Ah yes, good point :+1: